library/modules/EventManager.cpp

```diff
@@ -353,14 +372,15 @@ static void manageTickEvent(color_ostream& out) {
     }
     if ( toRemove.empty() )
         return;
-    for ( auto a = handlers[EventType::TICK].begin(); a != handlers[EventType::TICK].end(); ) {
-        EventHandler handle = (*a).second;
+    for (auto iter = handlers[EventType::TICK].begin(); iter != handlers[EventType::TICK].end(); ) {
+        EventHandler &handle = iter->second;
         if ( toRemove.find(handle) == toRemove.end() ) {
-            a++;
+            ++iter;
             continue;
         }
-        a = handlers[EventType::TICK].erase(a);
+        iter = handlers[EventType::TICK].erase(iter);
         toRemove.erase(handle);
+        registerTick(handle, handle.freq, iter->first);
         if ( toRemove.empty() )
             break;
     }
```

Comment:
What I am thinking is, that if we don't add to `handlers[TICK]` inside of `registerTick` then we can get both functionalities. Events which require re-registering a tick event handler, and regular listeners. If registered with the latter, we add it to the `handlers[TICK]` container, then in `manageTickEvent` we could, instead of iterating to remove a handle (which I think I may not have done correctly here) we could then iterate to re-queue it.

Reply:
@lethosor You probably have an order to your reviews, but in case you miss this. What are your thoughts?
```
int32_t DFHack::EventManager::registerTick(EventHandler handler, int32_t when, Plugin* plugin, bool absolute) {
    if ( !absolute ) {
        df::world* world = df::global::world;
        if ( world ) {
            when += world->frame_counter;
        } else {
            if ( Once::doOnce("EventManager registerTick unhonored absolute=false") )
                Core::getInstance().getConsole().print("EventManager::registerTick: warning! absolute flag=false not honored.\n");
        }
    }
    handler.when = when;
    tickQueue.insert(pair<int32_t, EventHandler>(handler.when, handler));
    eventLastTick[handler.eventHandler] = -1;
    //this commented line ensures "Registered Ticks" are not added back to the queue after execution
    //handlers[EventType::TICK].insert(pair<Plugin*,EventHandler>(plugin,handler));
    return when;
}
```

Reply:
What feedback are you looking for here?

Reply:
Well, basically whether to provide both mechanisms or not, but as discovered with diggingInvaders it is somewhat needed. Even if to avoid refactoring diggingInvaders a bunch.